hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/publish/TransactionPublisher.java

```diff
@@ -137,26 +150,33 @@ public void close() {
     }
 
     private Flux<Client> getClients() {
-        List<NodeProperties> validNodes = validateNodes();
-
-        if (validNodes.isEmpty()) {
-            throw new IllegalArgumentException("No valid nodes found");
-        }
+        validateNodes();
 
-        validNodes.forEach(n -> nodeAccountIds.add(AccountId.fromString(n.getAccountId())));
-        Map<String, AccountId> nodeMap = validNodes.stream()
+        log.info("Creating {} connections to {} nodes", publishProperties.getClients(), monitorProperties.getNodes()
+                .size());
+        Map<String, AccountId> nodes = monitorProperties.getNodes().stream()
                 .collect(Collectors.toMap(NodeProperties::getEndpoint, p -> AccountId.fromString(p.getAccountId())));
-        log.info("Creating {} connections to {} nodes", publishProperties.getClients(), validNodes.size());
 
+        NodeValidationProperties validationProperties = monitorProperties.getNodeValidation();
+        if (validationProperties.isEnabled()) {
+
+            nodeValidator = new AtomicReference<>(Flux.interval(validationProperties.getFrequency())
```

Comment:
The way you're using it is not atomic. Need to make the field final then:

```suggestion
            nodeValidator.compareAndSet(null, Flux.interval(validationProperties.getFrequency())
```

Reply:
Got it, fixed.